dropwizard-auth/src/main/java/io/dropwizard/auth/AuthDynamicFeature.java

```diff
@@ -0,0 +1,26 @@
+package io.dropwizard.auth;
+
+import com.google.common.base.Function;
+import org.glassfish.jersey.server.model.AnnotatedMethod;
+import javax.ws.rs.container.DynamicFeature;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class AuthDynamicFeature implements DynamicFeature {
+    final private AuthHandler authHandler;
+
+    public AuthDynamicFeature(AuthHandler authHandler) {
+        this.authHandler = authHandler;
+    }
+
+    @Override
+    public void configure(ResourceInfo resourceInfo, FeatureContext context) {
+        AnnotatedMethod am = new AnnotatedMethod(resourceInfo.getResourceMethod());
+
+        if (am.isAnnotationPresent(Auth.class)) {
```

Comment:
Given that we could be using `resourceInfo.getResourceMethod().getParameterAnnotations()`, why do we need to abandon the exiting Auth marker annotation? I admit, I don't know how we can go from `@Context SecurityContext securityContext` in the Resource parameter back to `@Auth SomeRandomAuthType authParam` to preserve backward compatibility, but this has always been my biggest concern in this change. 

Namely, is there anyway we can do this without having to force everyone to redo their auth code?